drivers/sensor/bmi323/bmi323_spi.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2023 Trackunit Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "bmi323.h"
+#include <zephyr/device.h>
+#include <zephyr/drivers/spi.h>
+
+static int bosch_bmi323_spi_read_words(const void *context, uint8_t offset, uint16_t *words,
+				       uint16_t words_count)
+{
+	const struct spi_dt_spec *spi = (const struct spi_dt_spec *)context;
+	uint8_t address[2];
+	struct spi_buf transmit_buffer;
+	struct spi_buf_set transmit_buffer_set;
+	struct spi_buf receive_buffers[2];
+	struct spi_buf_set receive_buffers_set;
+	int ret;
+
+	address[0] = offset | 0x80;
+	address[1] = 0x00;
+
+	transmit_buffer.buf = address;
+	transmit_buffer.len = sizeof(address);
+
+	transmit_buffer_set.buffers = &transmit_buffer;
+	transmit_buffer_set.count = 1;
+
+	receive_buffers[0].buf = NULL;
+	receive_buffers[0].len = 2;
+	receive_buffers[1].buf = words;
+	receive_buffers[1].len = (words_count * 2);
+
+	receive_buffers_set.buffers = receive_buffers;
+	receive_buffers_set.count = 2;
+
+	ret = spi_transceive_dt(spi, &transmit_buffer_set, &receive_buffers_set);
+
+	k_usleep(2);
```

Comment:
Why are all the spi transactions followed by a sleep?

Reply:
The BMI323 datasheet states that access to any register in any mode must be followed by an idle time of 2us. This sleep ensures the mutex which protects access to the BMI323 is held for at least 2us after the SPI transaction has transpired.

Since the sleep duration is lower than 1 tick on most embedded platforms, its preferable over keeping track of when the last transaction was made to perform a conditional block, which is the only other way I can think of to enforce the idle time otherwise :)